WordPress/src/main/java/org/wordpress/android/models/SiteSettingsModel.java

```diff
@@ -133,14 +136,20 @@ public boolean equals(Object other) {
         SiteSettingsModel otherModel = (SiteSettingsModel) other;
 
         return localTableId == otherModel.localTableId &&
-                address.equals(otherModel.address) &&
-                username.equals(otherModel.username) &&
-                password.equals(otherModel.password) &&
-                title.equals(otherModel.title) &&
-                tagline.equals(otherModel.tagline) &&
+                ((address == null && otherModel.address == null) ||
```

Comment:
These conditionals feels better than what it used to be, but not sure if it's necessary. Is there an issue that this is fixing? If so, should we separate it into it's own PR?

Also, assuming we keep these changes, is there a reason other objects don't use the same pattern like `defaultPostFormat` for example?

Again, assuming the changes are kept, do you think we should introduce a helper method to simplify this a bit? That way it'll be easier to read and understand and also apply it to other objects. What do you think?